machines/openIdBle/scan.ts

```diff
@@ -753,22 +754,31 @@ export const scanMachine =
       },
 
       services: {
-        checkBluetoothService: () => (callback) => {
-          const subscription = BluetoothStateManager.onStateChange((state) => {
-            if (state === 'PoweredOn') {
-              callback(model.events.BLUETOOTH_ENABLED());
+        checkBluetoothPermission: () => async (callback) => {
+          // wait a bit for animation to finish when app becomes active
+          await new Promise((resolve) => setTimeout(resolve, 250));
+          try {
+            const permission =
+              Platform.OS === 'ios'
+                ? PERMISSIONS.IOS.BLUETOOTH_PERIPHERAL
+                : PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION;
+            const response = await check(permission);
```

Comment:
Could we not check for any permission if its android, As this might show blueetooth permission error when location is disabled.